src/focus.service.ts

```diff
@@ -509,6 +509,22 @@ export class FocusService {
   private isFocusable(el: HTMLElement) {
     const role = el.getAttribute('role');
     const tabIndex = el.tabIndex;
+    let visible = el.offsetParent !== null;
+
+    for (let parent = el; parent !== this.root; parent = parent.parentElement) {
+      if (parent.classList.contains('hidden')) {
```

Comment:
Sorry, I'm not willing to approve a PR that has any form of this specific class checking in it. Checking styles is not terrible but this duct tape isn't really suitable for an open source library or one that's meant to be used in a complex application with more than a single developer working on it.